mlflow/models/evaluation/default_evaluator.py

```diff
@@ -332,6 +347,94 @@ def __validate_artifacts(artifacts):
     )
 
 
+def _load_custom_metric_artifact(
+    dataset_name, temp_dir, artifact_name, raw_artifact, custom_metric_index, custom_metric_name
+):
+    if isinstance(raw_artifact, EvaluationArtifact):
+        return raw_artifact
+
+    exception_header = _get_custom_metric_exception_header(custom_metric_index, custom_metric_name)
+
+    # Given local path, type inference through file extension
+    if isinstance(raw_artifact, str):
+        artifact_path = pathlib.Path(raw_artifact)
+        ext_to_artifact_map = {
+            ".png": ImageEvaluationArtifact,
+            ".jpg": ImageEvaluationArtifact,
+            ".jpeg": ImageEvaluationArtifact,
+            ".json": JsonEvaluationArtifact,
+            ".npy": NumpyEvaluationArtifact,
+            ".csv": CsvEvaluationArtifact,
+            ".parquet": ParquetEvaluationArtifact,
+        }
+        if artifact_path.is_file() and artifact_path.suffix in ext_to_artifact_map.keys():
+            # copy file to temp location with file name in the defined format
+            artifact_file_name = _gen_log_key(artifact_name, dataset_name) + artifact_path.suffix
+            artifact_file_local_path = temp_dir.path(artifact_file_name)
+            shutil.copyfile(raw_artifact, artifact_file_local_path)
+
+            mlflow.log_artifact(artifact_file_local_path)
+            return ext_to_artifact_map[artifact_path.suffix](
+                uri=mlflow.get_artifact_uri(artifact_file_name)
+            )
+        raise MlflowException(
+            f"{exception_header} produced an unsupported artifact '{artifact_name}' with path "
+            f"'{raw_artifact}'. The supported file extensions are:"
+            f".png, .jpg, .jpeg, .json, .npy, .csv, .parquet"
+        )
+
+    import matplotlib.pyplot as pyplot
+
+    # Given object, type inference
+    type_to_ext_map = {
+        pd.DataFrame: ".csv",
+        np.ndarray: ".npy",
+        pyplot.Figure: ".png",
+    }
+
+    type_to_artifact_map = {
+        pd.DataFrame: CsvEvaluationArtifact,
+        np.ndarray: NumpyEvaluationArtifact,
+        pyplot.Figure: ImageEvaluationArtifact,
+    }
+
+    artifact_file_name = _gen_log_key(artifact_name, dataset_name) + type_to_ext_map.get(
+        type(raw_artifact), ".json"
+    )
+
+    artifact_file_local_path = temp_dir.path(artifact_file_name)
+
+    if isinstance(raw_artifact, pd.DataFrame):
+        raw_artifact.to_csv(artifact_file_local_path, index=False)
+    elif isinstance(raw_artifact, np.ndarray):
+        np.save(artifact_file_local_path, raw_artifact, allow_pickle=False)
+    elif isinstance(raw_artifact, pyplot.Figure):
+        raw_artifact.savefig(artifact_file_local_path)
```

Comment:
This is probably out of scope for this PR, but have we discussed the option whether to save the entire figure object for interactive usage later? @WeichenXu123 @dbczumar